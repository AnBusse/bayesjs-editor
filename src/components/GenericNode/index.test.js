import React from 'react';
import { shallow } from 'enzyme';
import GenericNode from './component';

const defaultProps = {
  id: 'id',
  name: 'name',
  position: { x: 0, y: 0 },
  size: { height: 100, width: 100 },
  isSelected: false,
  onMouseDown: jest.fn(),
  onDoubleClick: jest.fn(),
  style: {},
  textX: 5,
  textY: 0,
  textHeight: 20,
  textWidth: 150,
  descriptionX: 5,
  descriptionY: 20,
  descriptionHeight: 20,
  descriptionWidth: 150,
  showDescription: false,
};

const shallowComponent = (props = {}) => {
  const compProps = { ...defaultProps, ...props };

  return shallow(<GenericNode {...compProps} />);
};

describe('GenericNode Component', () => {
  it('matches snapshot with description', () => {
    const component = shallowComponent({
      showDescription: true,
      descritpion: 'descritpion',
    });

    expect(component).toMatchSnapshot();
  });

  it('matches snapshot without description', () => {
    const component = shallowComponent();

    expect(component).toMatchSnapshot();
  });

  describe('Event Listeners', () => {
    describe('onMouseDown', () => {
      const onMouseDown = jest.fn();
      const component = shallowComponent({ onMouseDown });

      beforeEach(() => {
        component.simulate('mouseDown');
      });

      it('triggers "onMouseDown" on mouseDown event', () => {
        expect(onMouseDown).toBeCalled();
      });
    });

    describe('onDoubleClick', () => {
      const onDoubleClick = jest.fn();
      const component = shallowComponent({ onDoubleClick });

      beforeEach(() => {
        component.simulate('doubleClick');
      });

      it('triggers "onDoubleClick" on mouseDown event', () => {
        expect(onDoubleClick).toBeCalled();
      });
    });
  });
});
