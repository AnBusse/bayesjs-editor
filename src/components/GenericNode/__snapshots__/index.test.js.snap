// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericNode Component matches snapshot with description 1`] = `
<g
  className="node"
  data-testid="generic-node__id"
  onDoubleClick={[Function]}
  onMouseDown={[Function]}
  style={Object {}}
  transform="translate(0 0)"
>
  <rect
    fill="#ff8"
    fillOpacity="1"
    height={100}
    stroke="#333"
    strokeWidth={1}
    width={100}
  />
  <SvgText
    height={20}
    text="name"
    width={150}
    x={5}
    y={0}
  />
  <NodeDivider
    width={100}
    y={20}
  />
  <SvgText
    height={20}
    text={null}
    width={150}
    x={5}
    y={20}
  />
</g>
`;

exports[`GenericNode Component matches snapshot without description 1`] = `
<g
  className="node"
  data-testid="generic-node__id"
  onDoubleClick={[Function]}
  onMouseDown={[Function]}
  style={Object {}}
  transform="translate(0 0)"
>
  <rect
    fill="#ff8"
    fillOpacity="1"
    height={100}
    stroke="#333"
    strokeWidth={1}
    width={100}
  />
  <SvgText
    height={20}
    text="name"
    width={150}
    x={5}
    y={0}
  />
  <NodeDivider
    width={100}
    y={20}
  />
</g>
`;
